crates/sargon/src/profile/mfa/security_structures/roles/factor_levels/factor_instance_level/role_into_scrypto_access_rule.rs

```diff
@@ -0,0 +1,56 @@
+use crate::prelude::*;
+
+impl<const ROLE: u8> From<RoleWithFactorInstances<ROLE>> for ScryptoAccessRule {
+    fn from(value: RoleWithFactorInstances<ROLE>) -> Self {
+        let from_factors =
+            |factors: &Vec<FactorInstance>| -> Vec<ScryptoResourceOrNonFungible> {
+                factors
+                    .iter()
+                    .map(|instance| instance.badge.clone())
+                    .map(ScryptoResourceOrNonFungible::from)
+                    .collect()
+            };
+        ScryptoAccessRule::Protected(ScryptoCompositeRequirement::AnyOf(vec![
+            ScryptoCompositeRequirement::BasicRequirement(
+                ScryptoBasicRequirement::CountOf(
+                    value.get_threshold(),
+                    from_factors(value.get_threshold_factors()),
+                ),
+            ),
+            ScryptoCompositeRequirement::BasicRequirement(
+                ScryptoBasicRequirement::AnyOf(from_factors(
+                    value.get_override_factors(),
+                )),
+            ),
+        ]))
+    }
+}
+
+impl From<MatrixOfFactorInstances> for ScryptoRuleSet {
+    fn from(
+        MatrixOfFactorInstances {
+            primary_role,
+            recovery_role,
+            confirmation_role,
+            ..
+        }: MatrixOfFactorInstances,
+    ) -> Self {
+        Self {
+            primary_role: primary_role.into(),
+            recovery_role: recovery_role.into(),
+            confirmation_role: confirmation_role.into(),
+        }
+    }
+}
+
+pub const MINUTES_PER_DAY: u32 = 24 * 60;
```

Comment:
We do have a constants file

Reply:
constants which have a "natural home" - should be put in a natural home I think. That is most Rusty I think. The constants file seems to contain mostly constants relating to GW?

Reply:
well I agree, at the same time `MINUTES_PER_DAY` is very generic. Anyway, this is minor, we'll migrate in case there will be other components using this.

Reply:
right, ok agreed, this should be moved, since it is common relating to time!